src/task/blocking.rs

```diff
@@ -47,18 +58,34 @@ lazy_static! {
     };
 }
 
-// Create up to MAX_THREADS dynamic blocking task worker threads.
-// Dynamic threads will terminate themselves if they don't
-// receive any work after between one and ten seconds.
-fn maybe_create_another_blocking_thread() {
-    // We use a `Relaxed` atomic operation because
-    // it's just a heuristic, and would not lose correctness
-    // even if it's random.
-    let workers = DYNAMIC_THREAD_COUNT.load(Ordering::Relaxed);
-    if workers >= MAX_THREADS {
-        return;
+fn calculate_dispatch_frequency() {
+    // Calculate current message processing rate here
+    let current_freq = FREQUENCY.fetch_sub(1, Ordering::Relaxed);
```

Comment:
This only happens to be properly serialized due to the SeqCst barrier placed on the CURRENT_POOL_SIZE load in schedule(), but if that changes this is vulnerable to an underwrap. Please make it at least `Ordering::Acquire` for defense.